tests/integration/sqs_work_dispatcher_tests.py

```diff
@@ -0,0 +1,522 @@
+            else:
+                return two_work, (message.body,)
+
+        dispatcher.dispatch_by_message_attribute(work_one_or_two)
+        dispatcher._worker_process.join(5)  # wait at most 5 sec for the work to complete
+
+        # Make sure the "a_work" was done
+        messages = queue.receive_messages(WaitTimeSeconds=1, MaxNumberOfMessages=10)
+        self.assertEqual(1, len(messages))
+        self.assertEqual(1, messages[0].body)
+
+        # Worker process should have a successful (0) exitcode
+        self.assertEqual(0, dispatcher._worker_process.exitcode)
+
+    def test_faulty_queue_connection_raises_correct_exception(self):
+        """When a queue cannot be connected to, it raises the appropriate exception"""
+        try:
+            # note: connection max retries config not in botocore v1.5.x
+            client_config = Config(region_name="us-gov-west-1", connect_timeout=1, read_timeout=1)
+            sqs = boto3.resource('sqs', config=client_config)
+            queue = sqs.Queue("75f4f422-3866-4e4f-9dc9-5364e3de3eaf")
+            dispatcher = SQSWorkDispatcher(queue, worker_process_name="Test Worker Process",
+                                           long_poll_seconds=1, monitor_sleep_time=1)
+            dispatcher.dispatch(lambda x: x*2)
+        except (SystemExit, Exception) as e:
+            self.assertIsInstance(e, SystemExit)
+            self.assertIsNotNone(e.__cause__)
+            self.assertTrue(isinstance(e.__cause__, EndpointConnectionError) or
+                            isinstance(e.__cause__, ClientError) or
+                            isinstance(e.__cause__, NoRegionError) or
+                            isinstance(e.__cause__, NoCredentialsError))
+
+    def test_failed_job_detected(self):
+        """SQSWorkDispatcher handles failed work within the child process
+
+        - Given a numeric message body
+        - When on a SQSWorkDispatcher.dispatch() is called
+        - And the function to execute in the child process fails
+        - Then a QueueWorkerProcessError exception is raised
+        - And the exit code of the worker process is > 0
+        """
+        with self.assertRaises(QueueWorkerProcessError):
+            queue = sqs_queue()
+            queue.send_message(MessageBody=1234)
+
+            dispatcher = SQSWorkDispatcher(queue, worker_process_name="Test Worker Process",
+                                           long_poll_seconds=1, monitor_sleep_time=1)
+
+            def fail_at_work(task_id):
+                raise Exception("failing at this particular job...")
+
+            dispatcher.dispatch(fail_at_work)
+            dispatcher._worker_process.join(5)  # wait at most 5 sec for the work to complete
+
+        # Worker process should have a failed (> 0)  exitcode
+        self.assertGreater(dispatcher._worker_process.exitcode, 0)
+
+    @unittest.skip("Test is not provable with asserts. Reading STDOUT proves it, but a place to store shared state "
```

Comment:
If the assertable portion of this test is still good, I don't see why we should skip it. My suggestion would just be to include a note explaining the latter half of the test.

Reply:
Done
Removed skip